ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -957,6 +1043,163 @@ def wait_until_cpu_port_up(self):
+            return None
+        # Filter out packets and remove floods:
+        self.unique_id = list()     # This list will contain all unique Payload ID, to filter out received floods.
+        filtered_packets = [ pkt for pkt in all_packets if \
+            scapyall.UDP in pkt and \
+            not scapyall.ICMP in pkt and \
+            pkt[scapyall.UDP].sport == 1234 and \
+            pkt[scapyall.UDP].dport == 5000 and \
+            self.check_udp_payload(pkt) and \
+            self.no_flood(pkt)
+            ]
+        # Re-arrange packets, if delayed, by Payload ID and Timestamp:
+        packets = sorted(filtered_packets, key = lambda packet: (int(str(packet[scapyall.UDP].payload)), packet.time ))
+        self.lost_packets = dict()
+        self.max_disrupt, self.total_disruption = 0, 0
+        sent_packets = dict()
+        if packets:
+            prev_payload, prev_time = 0, 0
+            sent_payload = 0
+            received_counter = 0
+            self.disruption_start, self.disruption_stop = None, None
+            for packet in packets:
+                if packet[scapyall.Ether].dst == self.dut_mac:
+                    # This is a sent packet - keep track of it as payload_id:timestamp.
+                    sent_payload = int(str(packet[scapyall.UDP].payload))
+                    sent_packets[sent_payload] = packet.time
+                    continue
+                if packet[scapyall.Ether].src == self.dut_mac:
+                    # This is a received packet.
+                    received_time = packet.time
+                    received_payload = int(str(packet[scapyall.UDP].payload))
+                    received_counter += 1
+                if not (received_payload and received_time):
+                    # This is the first valid received packet.
+                    prev_payload = received_payload
+                    prev_time = received_time
+                    continue
+                if received_payload - prev_payload > 1:
+                    # Packets in a row are missing, a disruption.
+                    lost_id = (received_payload -1) - prev_payload # How many packets lost in a row.
+                    disrupt = (sent_packets[received_payload] - sent_packets[prev_payload + 1]) # How long disrupt lasted.
+                    # Add disrupt to the dict:
+                    self.lost_packets[prev_payload] = (lost_id, disrupt, received_time - disrupt, received_time)
+                    self.log("Disruption between packet ID %s and %s. For %s " % (prev_payload, received_payload, disrupt))
+                    if not self.disruption_start:
+                        self.disruption_start = datetime.datetime.fromtimestamp(prev_time)
+                    self.disruption_stop = datetime.datetime.fromtimestamp(received_time)
+                prev_payload = received_payload
+                prev_time = received_time
+        else:
+            self.log("Packets list is empty")
```

Comment:
Should this be a failure?

Reply:
Added the failure before the analyzer:
self.assertTrue(packets, "Sniffer failed to capture any traffic")

Added the faulure after the anayzer:
self.assertTrue(received_counter, "Sniffer failed to filter any traffic from DUT")

And removed redundant 'else' clauses.